crates/accelerate/src/unitary_synthesis.rs

```diff
@@ -565,28 +632,15 @@ fn get_2q_decomposers_from_target(
     }
 
     #[inline]
-    fn replace_parametrized_gate(mut op: NormalOperation) -> NormalOperation {
+    fn check_parametrized_gate(op: NormalOperation) -> bool {
         if let Some(std_gate) = op.operation.try_standard_gate() {
-            match std_gate.name() {
-                "rxx" => {
-                    if let Param::ParameterExpression(_) = op.params[0] {
-                        op.params[0] = Param::Float(PI2)
-                    }
-                }
-                "rzx" => {
-                    if let Param::ParameterExpression(_) = op.params[0] {
-                        op.params[0] = Param::Float(PI4)
-                    }
+            if PARAM_SET.contains(&std_gate.name()) {
+                if let Param::ParameterExpression(_) = op.params[0] {
+                    return true;
```

Comment:
I think we might be missing out a group of gates here. As far as I understand it, the `TwoQubitControlledUDecomposer` works not only for standard gates (the ones defined in `PARAM_SET`), but should also be at least a candidate for user-defined parametrized unitary gates that match the input conditions. In that case, we should probably not check `PARAM_SET.contains(&std_gate.name())`, and instead consider any 2q unitary parametrized gate for the "`available_2q_param_basis`".

Reply:
Note that not any parametrized gate can fit here, only certain parametrized gates which are equivalent (up to single qubit gates) to an `RXXGate`. The `TwoQubitControlledUDecomposer` works for such gates if the user was able to calibrate them somehow. Should the `UnitarySynthesis` transpiler pass fit any calibrated gates?
So I'm not sure what exactly should be checked here. See the function `TwoQubitControlledUDecomposer.to_rxx_gate`

Reply:
We currently support non-standard gates too, so the pass should work if there is a non-standard gate that meets the condition of rxx equivalence. `TwoQubitControlledUDecomposer::new` already checks if the gate is rxx equivalent:https://github.com/Qiskit/qiskit/blob/e9ccd3f374fd5424214361d47febacfa5919e1e3/crates/accelerate/src/two_qubit_decompose.rs#L2777-L2822

So I was thinking that we could abstract that logic and add a check to `UnitarySynthesis` similar to `is_controlled` or `is_supercontrolled`, but in this case it could be called `is_rxx_equivalent`(for example). The idea for the check is to filter out the non-rxx-equivalent gates.